pkg/kv/kvserver/logstore/bytes_tracker.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package logstore
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
+)
+
+// MaxRaftEntriesSize is the "global" soft limit for the total size of raft log
+// entries pulled into memory simultaneously. Currently, this only includes the
+// entries pulled as part of the local state machine application flow.
+//
+// TODO(pav-kv): track/limit usage for other large memory consumers in raft:
+//   - the leader->follower append flows,
+//   - the raft/unstable data structure.
+var MaxRaftEntriesSize = settings.RegisterByteSizeSetting(
```

Comment:
A couple of concerns with setting this limit via a cluster setting:
- There is a bit of a circular dependency. Raft flows depend on this setting, and the setting changes are committed to state machine via raft.
- Having this global setting assumes all servers have similar capacities, which isn't necessarily true.

Maybe we should only control enable/disable via a cluster setting, and make the actual limit more static / per-server.

Reply:
Made it an env var.